molgenis-security/src/main/java/org/molgenis/security/oidc/MappedOidcUserService.java

```diff
@@ -1,37 +1,72 @@
 package org.molgenis.security.oidc;
 
 import static java.util.Objects.requireNonNull;
+import static org.springframework.security.oauth2.core.oidc.StandardClaimNames.SUB;
 
 import java.util.HashSet;
 import java.util.Set;
+import org.molgenis.data.DataService;
+import org.molgenis.data.UnknownEntityException;
 import org.molgenis.data.security.auth.User;
+import org.molgenis.security.core.runas.RunAsSystem;
+import org.molgenis.security.oidc.model.OidcClient;
+import org.molgenis.security.oidc.model.OidcClientMetadata;
 import org.molgenis.security.user.UserDetailsServiceImpl;
 import org.springframework.security.core.GrantedAuthority;
 import org.springframework.security.oauth2.client.oidc.userinfo.OidcUserRequest;
 import org.springframework.security.oauth2.client.oidc.userinfo.OidcUserService;
+import org.springframework.security.oauth2.client.userinfo.OAuth2UserService;
 import org.springframework.security.oauth2.core.oidc.user.OidcUser;
 
-public class MappedOidcUserService extends OidcUserService {
+public class MappedOidcUserService implements OAuth2UserService<OidcUserRequest, OidcUser> {
```

Comment:
Instead of extending, delegate. This way we can mock the way the actual user service loads the user.